app/demo-geant-integration/HitRootIO.cc

```diff
@@ -170,13 +179,29 @@ void HitRootIO::Close()
  */
 void HitRootIO::Merge()
 {
-    auto nthreads = G4RunManager::GetRunManager()->GetNumberOfThreads();
+#if G4VERSION_NUMBER >= 1070
+    auto const nthreads = G4RunManager::GetRunManager()->GetNumberOfThreads();
+#else
+    auto const nthreads = std::invoke([]() {
```

Comment:
I like the IIFE, but you should take it one step further and move the compiler macro inside it so we don't have to declare the variable twice 😄

Reply:
yes, I was considering that but ended up doing it that way for readability

Reply:
Are you saying that you find:
```c++
#if G4VERSION_NUMBER >= 1070
    auto const nthreads = G4RunManager::GetRunManager()->GetNumberOfThreads();
#else
    auto const nthreads = std::invoke([]() {
```
to be more readable than
```c++
    auto const nthreads = std::invoke([]() {
#if G4VERSION_NUMBER >= 1070
        return G4RunManager::GetRunManager()->GetNumberOfThreads();
#else
```
?

Reply:
I was thinking of
```c++
auto const nthreads =
#if G4VERSION_NUMBER >= 1070
    G4RunManager::GetRunManager()->GetNumberOfThreads();
#else
    std::invoke([]() {});
#endif
```
In any case, if I move the lambda to a free function we no longer need the IIFE

Reply:
That sort of syntax can mess with clang-format... it's better IMHO to use preprocessor macros to comment out entire statements, not fragments of one.

Reply:
Well, now we have to declare it twice unless we want to use a lambda just to avoid it as you suggested before?